PurchasesTests/Purchasing/OfferingsManagerTests.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  OfferingsManagerTests.swift
+//
+//  Created by Juanpe Catalán on 9/8/21.
+
+import XCTest
+import Nimble
+import StoreKit
+import PurchasesCoreSwift
+
+// TODO (post-migration): Move this test case to the new framework target (PurchasesCoreSwift).
+class OfferingsManagerTests: XCTestCase {
+
+    let mockDeviceCache = MockDeviceCache()
+    let mockOperationDispatcher = MockOperationDispatcher()
+    let mockSystemInfo = try! MockSystemInfo(platformFlavor: "iOS",
+                                             platformFlavorVersion: "3.2.1",
+                                             finishTransactions: true)
+    let mockBackend = MockBackend()
+    let mockOfferingsFactory = MockOfferingsFactory()
+    let mockProductsManager = MockProductsManager()
+    var offeringsManager: OfferingsManager!
+
+    override func setUp() {
+        super.setUp()
+        
+        offeringsManager = OfferingsManager(deviceCache: mockDeviceCache,
+                                            operationDispatcher: mockOperationDispatcher,
+                                            systemInfo: mockSystemInfo,
+                                            backend: mockBackend,
+                                            offeringsFactory: mockOfferingsFactory,
+                                            productsManager: mockProductsManager)
+    }
+
+}
+
+extension OfferingsManagerTests {
+
+    func testMissingProductDetailsReturnsNil() {
+        // given
+        mockOfferingsFactory.emptyOfferings = true
+        mockBackend.stubbedGetOfferingsCompletionResult = (MockData.anyBackendOfferingsData, nil)
+
+        // when
+        var obtainedOfferings: Offerings?
+        offeringsManager.offerings(appUserID: MockData.anyAppUserID) { offerings, _ in
+            obtainedOfferings = offerings
+        }
+
+        // then
+        expect(obtainedOfferings).toEventuallyNot(beNil());
```

Comment:
couple of notes here: 

For tests like this one, I've ran into a few situations where a test breaks, and it's hard to tell what the problem is. 

For example, in this case, if the `obtainedOfferings` object is `nil`, it could happen both because `completion` was never called, and because it was called, but with `offerings == nil`. 

What I've been doing to get around it is to add a variable called `completionCalled`, and use that to wait for the completion. 

Also, once the completion _has_ been called, the other checks don't have to use `toEventually`, they can just check against the current value 🎉 

i.e.: 

```swift

// when
var obtainedOfferings: Offerings?
var completionCalled = false
offeringsManager.offerings(appUserID: MockData.anyAppUserID) { offerings, _ in
    obtainedOfferings = offerings
    completionCalled = true
}

// then
expect(completionCalled).toEventually(beTrue())
expect(obtainedOfferings).toNot(beNil())
```
Also, there's an extra `;` at the end of the line

Reply:
also, I realize that these tests existed before this PR, but I hadn't actually read them through yet 😂

Reply:
Yeah, some of these tests were only moved to this file 🙂

I've updated the methods checking if the completion was called. It's much better 👍🏼